app/views/dashboard/works/edit.html.erb

```diff
@@ -124,6 +124,28 @@
       </div>
 
       <% if current_user.admin? && policy(@work.latest_version).destroy? %>
```

Comment:
Initially I put curator in that block because that's where the depositor form is & they have the same access/permissions requirements.

Digging deeper, I don't think either curator or depositor should be there.  It looks like the full if condition [was first added](https://github.com/psu-libraries/scholarsphere/issues/867) to encompass the 'Danger Zone' so checking admin & able to delete made sense. Then it was requested that [the depositor form be added](https://github.com/psu-libraries/scholarsphere/issues/1031) to that zone & it was added within the same condition block, but not within the danger section.

So I think curator, depositor, and danger should be within `if current_user.admin?` but `if policy(@work.latest_version).destroy?` should just be around Lines 157-179 (danger section about deleting). Does that sound right?

Reply:
That certainly sounds right to me, but I'm new here 😄

Reply:
The logic in the WorkVersionPolicy for destroy? returns false even for admins if the version is not the latest or the version is not persisted:
```
  def destroy?
    return false if record.new_record?
    return true if user.admin? && record == record.work.latest_version

    !record.published? && editable?
  end
```

Perhaps there was a reason to prevent editing the depositors in these cases?  I wouldn't change the logic for the depositors, but we might want the curator logic outside of that?

Reply:
Although, I guess we're always passing the latest version in here.

Reply:
> Perhaps there was a reason to prevent editing the depositors in these cases?

Yeah, good question. Is there a test for this logic somewhere? If so, then I think we can assume that it was intentional. If not, 🤷‍♂️... who would know?

Reply:
The original [request](https://github.com/psu-libraries/scholarsphere/issues/1031) for that feature sounds like Seth was describing things in terms of how the UI looks and not necessarily in terms of how the business logic should work. It sounds like the important thing is that only admins can change depositors (which is also true for the "danger zone" actions). There might have been a miscommunication there. On the other hand, there doesn't seem to be a problem with the way that it has been working up to now.

If the way that it's working now is intentional and correct, then there's probably a better name than "destroy?" for that policy..

Reply:
I think we should just leave things as they are and wait to see if any issues pop up.  I agree, though, that the policy method's name (the way we are using it here) doesn't quite describe what's going on.  I don't see any tests that test for any of this logic other than admin vs non-admin.